src/viem/saga.ts

```diff
@@ -74,53 +82,117 @@ export function* sendPayment({
     yield* call(unlockAccount, wallet.account.address)
 
     yield* put(
       addStandbyTransaction({
         context,
-        networkId: networkConfig.defaultNetworkId,
+        networkId: tokenInfo.networkId,
         type: TokenTransactionTypeV2.Sent,
         comment,
         status: TransactionStatus.Pending,
         value: amount.negated().toString(),
-        tokenAddress,
-        tokenId: getTokenId(networkConfig.defaultNetworkId, tokenAddress),
+        tokenAddress: tokenInfo.address ?? undefined,
+        tokenId,
         timestamp: Math.floor(Date.now() / 1000),
         address: recipientAddress,
       })
     )
+  }
 
-    const receipt = yield* call(sendAndMonitorTransaction, {
-      context,
-      wallet,
-      // Cast for now otherwise TS complains about the type of request
-      // TODO: investigate more and fix
-      request: request as SimulateContractReturnType['request'],
-    })
-    return receipt
+  const tokenAddress = tokenInfo.address
+
+  // For tokens with an address, we simulate calling 'transfer' on the contract,
+  // take the request generated by that simulation, and execute that request
+  //
+  // For tokens with no address, we perform a simple `call` to test the request.
+  try {
+    if (tokenAddress) {
+      if (!feeInfo) {
+        throw new Error('non-native viem sends require feeInfo')
+      }
+      // this returns a method which is then passed to call instead of directly
+      // doing yield* call(publicClient.celo.simulateContract, args) because this
+      // results in a long TS error
+      const simulateContractMethod = yield* call(getTransferSimulateContract, {
+        wallet,
+        tokenId,
+        amount,
+        recipientAddress,
+        comment,
+        feeInfo,
+      })
+
+      const { request } = yield* call(simulateContractMethod)
+      yield* call(unlockAndAddStandby)
+
+      const sendContractTxMethod = function* () {
+        return yield* call(
+          [wallet, 'writeContract'],
+          request as SimulateContractReturnType['request']
+        )
+      }
+
+      const receipt = yield* call(sendAndMonitorTransaction, {
+        context,
+        network,
+        sendTx: sendContractTxMethod,
+      })
+
+      return receipt
+    } else {
+      const convertedAmount = BigInt(
+        yield* call(tokenAmountInSmallestUnit, amount, tokenInfo.tokenId)
+      )
+
+      // The assumption here is that if the call throws, then the actual execution would result
+      // in a revert. I'm not positive that this assumption holds, however.
+      // TODO (ACT-922): Fill in fee estimation for native token transfers (non-Celo)
+      const callMethod = async () =>
+        await publicClient[network].call({
+          account: wallet.account,
+          to: getAddress(recipientAddress),
+          data: '0x00',
+          value: convertedAmount,
+        })
+      Logger.info(TAG, 'Call completed successfully')
+      const callData = yield* call(callMethod)
```

Comment:
I've not seen `callData` end up as anything other than an empty object. The [call docs](https://viem.sh/docs/actions/public/call.html) mark `data` as a required field, but I've been able to omit `data` without TS yelling at me. I'm not convinced that this is how we ought to simulate native sends, or if it's possible(???) using Veim?